hnn_core/cell.py

```diff
@@ -202,6 +204,14 @@ def __init__(self, L, diam, Ra, cm, end_pts=None):
         self.mechs = dict()
         self.syns = list()
 
+        # For distance functionality
+        self.nseg = 1
+        if self.L > 100.:  # 100 um
+            self.nseg = int(self.L / 50.)
+            # make dend.nseg odd for all sections
+            if not self.nseg % 2:
+                self.nseg += 1
```

Comment:
you need to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) your code:

https://github.com/raj1701/hnn-core/blob/neuron_coordinates/hnn_core/cell.py#L504-L508

make a private function `_get_nseg` that you use in both parts of the code.